weblate/memory/models.py

```diff
@@ -280,6 +294,70 @@ def import_tmx(
                 found += 1
         return found
 
+    def import_other_format(
+        self,
+        request,
+        fileobj,
+        origin,
+        source_language: str | None = None,
+        target_language: str | None = None,
+        **kwargs,
+    ) -> int:
+        """
+        Import memory from other formats.
+
+        This is a generic function to import memories from other formats.
+        It currently supports all formats supported by `try_load` from
+        `weblate.formats.auto`.
+
+        """
+        from weblate.formats.auto import try_load
+
+        langcache = {}
+        try:
+            storage = try_load(origin, fileobj.read(), None, None)
+        except Exception as error:
+            report_error("Could not parse memory")
+            raise MemoryImportError(gettext("Unsupported file!")) from error
+
+        if storage.monolingual is True:
+            raise MemoryImportError(
+                gettext("Monolingual format not supported for memory upload")
+            )
+
+        def get_language(language: Language | str | None) -> Language:
+            """Get a language object based on the given code."""
+            if isinstance(language, Language):
+                return language
+
+            if not language:
+                raise MemoryImportError(
+                    gettext("Missing source or target language in file!")
+                )
+            try:
+                return Language.objects.get_by_code(language, langcache)
+            except Language.DoesNotExist as error:
+                raise MemoryImportError(
+                    gettext("Could not find language %s!") % language
+                ) from error
+
+        source_language = get_language(storage.source_language or source_language)
+        target_language = get_language(storage.language_code or target_language)
+
+        count = 0
+        for unit in storage.all_units:
```

Comment:
Filtering for translated units only should happen here as well. Maybe using `iterate_merge` would be a better method (it is used for translations import).